usagov/vote-gov-drupal
# VOTE-576 NVRF data model and api refinements

Eligibility placeholders
Embedded iconlist
NVRF PDF pages
reg_type refinement
mail_deadline refinement



## Jira ticket

https://cm-jira.usa.gov/browse/VOTE-576

## Description

NVRF data model and api refinements

## Deployment and testing

### Post-deploy

1. run `lando retune`

### QA/Test

1. visit http://vote-gov.lndo.site/node/69/edit and add a placeholder with machine name  "reg_eligibility_desc" using the "Embedded content" button. It should display as @reg_eligibility_desc in the wysiwyg and also in the api endpoint http://vote-gov.lndo.site/api/nvrf/pages.json
2. visit http://vote-gov.lndo.site/node/7/edit in the NVRF tab there is a "NVRF PDF pages" field at the end of the field list. Add "0,1,2,3,4,5" then visit the api link to see it as a new property in the state http://vote-gov.lndo.site/api/nvrf/states.json
3. visit http://vote-gov.lndo.site/api/nvrf/states.json and see that the "reg_type" property has only 1 value
4. visit http://vote-gov.lndo.site/api/nvrf/states.json and see that general mail deadlines are removed and combined with the deadline value as a fallback. Edit http://vote-gov.lndo.site/node/7/edit and add a specific postmaked mail deadline then go back to the api link and see that the specific deadline overrides the general one.

## Checklist for the Developer




## Checklist for the Peer Reviewers
